### PR TITLE
[Merged by Bors] - bump operator-rs to 0.27.1

### LINKED ISSUE #344: Bump operator-rs to 0.27.1
Update operator-rs to [0.27.1](https://github.com/stackabletech/operator-rs/blob/main/CHANGELOG.md).

- [x] operator-rs bumped
- [x] operator has no dependecy on kube or serde_yaml (use https://github.com/stackabletech/operator-rs/pull/450 instead)
- [x] Use `Fragment` wherever possible
- [x] Orphaned resources mechanism is used
- [x] Use parser for k9s `Quantity` instead of parsing memory values yourself

Product image selection will be tracked later on by https://github.com/stackabletech/issues/issues/305 but should be pretty easy compared to the changes in this Issue

### PR DESCRIPTION
**Pull request by adwk67**
# Description

Fixes #344.
:heavy_check_mark: CI: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/druid-operator-it-custom/43/

## Review Checklist

- [x] Code contains useful comments
- [x] CRD change approved (or not applicable)
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: deploy/manifests/crds.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ---
+---
```
Comment (razvan) [inline deploy/manifests/crds.yaml:2]: Something went wrong here.

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -30,7 +31,7 @@ use std::{
 use strum::{Display, EnumDiscriminants, EnumIter, EnumString, IntoStaticStr};
 
 pub const APP_NAME: &str = "druid";
-pub const CONTROLLER_NAME: &str = "druid-operator";
+pub const OPERATOR_NAME: &str = "airflow.stackable.tech";
```
Comment (razvan) [inline rust/crd/src/lib.rs:34]: The copy-pasta police disapproves!

**Review by razvan — COMMENTED** [review]
The CHANGELOG entry is missing.

File: rust/crd/src/lib.rs (hunk 2 above)
Comment (adwk67) [reply rust/crd/src/lib.rs:34]: whoops :(

File: deploy/manifests/crds.yaml (hunk 1 above)
Comment (maltesander) [reply deploy/manifests/crds.yaml:2]: Yeah i have the same with HBase, not sure where this is coming from?

File: rust/crd/src/lib.rs (hunk 2 above)
Comment (adwk67) [reply rust/crd/src/lib.rs]: Fixed.

File: deploy/manifests/crds.yaml (hunk 1 above)
Comment (razvan) [reply deploy/manifests/crds.yaml:2]: we want to get rid of these anyway.

**Comment by adwk67** [comment]
bors merge

**Pull request closed without merging**